datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -318,9 +433,12 @@ impl Stream for GroupedHashAggregateStream {
                             assert!(!self.input_done);
 
                             if let Some(to_emit) = self.group_ordering.emit_to() {
-                                let batch = extract_ok!(self.emit(to_emit));
+                                let batch = extract_ok!(self.emit(to_emit, false));
                                 self.exec_state = ExecutionState::ProducingOutput(batch);
                             }
+
+                            extract_ok!(self.emit_early_if_necessary());
```

Comment:
Good point, I was also thinking about this. I ended up choosing to keep the if conditions at the inner scope in order to keep the outer scope clean as much as possible. It is going to be likely I remove the `matches!(self.group_ordering, GroupOrdering::None)` from `spill_previous_if_necessary` and `emit_early_if_necessary`, and so that the changes are only in those functions. But largely TBD